### PR TITLE
Add 2 error code for canary new built-in roles

### PR DESCRIPTION
**Pull request by Sodaine**
Roadmap work item id : 17660

### CODE REVIEW

**Review by RuiHou105 — CHANGES_REQUESTED** [review]
@Sodaine  Please:
1.Update the changelog to describe your change.
2.Create roadmap workitem and attach the workitem number in comments.

**Comment by Sodaine** [comment]
> @Sodaine Please: 1.Update the changelog to describe your change. 2.Create roadmap workitem and attach the workitem number in comments.

Done

**Comment by Sodaine** [comment]
> @Sodaine Please: 1.Update the changelog to describe your change. 2.Create roadmap workitem and attach the workitem number in comments.

Done

**Review by RuiHou105 — APPROVED** [review]

**Comment by Danielabom** [comment]
@Jarbas-MSFT @Lauragra, moving this PR back into my queue.

File: changelog/Microsoft.CloudManagedDesktop.json [hunk 1]
```diff
@@ -3188,6 +3188,32 @@
...
-      }
+    },
+    {
+        "ChangeList": [
+            {
+                "Id": "a646af91-1226-4824-9e44-e91b7d27ef4a",
+                "ApiChange": "Property",
+                "ChangedApiName": "permissionCheckNoWindows365NetworkUserRole",
+                "ChangeType": "Addition",
+                "Description": "Added the **permissionCheckNoWindows365NetworkUserRole** member to the [cloudPcOnPremisesConnectionHealthCheckErrorType](https://learn.microsoft.com/en-us/graph/api/resources/cloudpconpremisesconnectionhealthcheck?view=graph-rest-beta#cloudpconpremisesconnectionhealthcheckerrortype-values) resource.",
```
Comment (Danielabom) [inline changelog/Microsoft.CloudManagedDesktop.json:3199]:
```suggestion
                "Description": "Added the `permissionCheckNoWindows365NetworkUserRole` member to the **cloudPcOnPremisesConnectionHealthCheckErrorType** enumeration .",
```

File: changelog/Microsoft.CloudManagedDesktop.json [hunk 2]
```diff
@@ -3188,6 +3188,32 @@
...
+                "Target": "cloudPcOnPremisesConnectionHealthCheckErrorType"
+            },
+            {
+                "Id": "a646af91-1226-4824-9e44-e91b7d27ef4a",
+                "ApiChange": "Property",
```
Comment (Danielabom) [inline changelog/Microsoft.CloudManagedDesktop.json:3204]:
```suggestion
                "ApiChange": "Member",
```

File: changelog/Microsoft.CloudManagedDesktop.json [hunk 3]
```diff
@@ -3188,6 +3188,32 @@
...
+                "ChangedApiName": "permissionCheckNoWindows365NetworkInterfaceContributorRole",
+                "ChangeType": "Addition",
+                "Description": "Added the **permissionCheckNoWindows365NetworkInterfaceContributorRole** member to the [cloudPcOnPremisesConnectionHealthCheckErrorType](https://learn.microsoft.com/en-us/graph/api/resources/cloudpconpremisesconnectionhealthcheck?view=graph-rest-beta#cloudpconpremisesconnectionhealthcheckerrortype-values) resource.",
```
Comment (Danielabom) [inline changelog/Microsoft.CloudManagedDesktop.json:3207]:
```suggestion
                "Description": "Added the `permissionCheckNoWindows365NetworkInterfaceContributorRole` member to the **cloudPcOnPremisesConnectionHealthCheckErrorType** enumeration.",
```

File: api-reference/beta/resources/cloudpconpremisesconnectionhealthcheck.md [hunk 4]
```diff
@@ -90,6 +90,8 @@ The result of a Cloud PC Azure network connection health check.
...
 |permissionCheckNoVNetContributorRole|The Cloud PC service principal doesn't have sufficient permissions on the Azure vNet. Please make sure that the Cloud PC service has the *Network contributor* permissions on the vNet.|
 |permissionCheckNoResourceGroupNetworkContributorRole|Cloud PC service principal doesn't have sufficient permissions on the Azure resource group. Please make sure that the  application has Network contributor permissions on the resource group.|
+|permissionCheckNoWindows365NetworkUserRole|Cloud PC service principal does not have Windows365 network user role on the specified virtual network. Please work with subscription owner to add the  Windows365 network user role assignment for the Cloud PC service principal.|
+|permissionCheckNoWindows365NetworkInterfaceContributorRole|Cloud PC service principal does not have Windows365 network interface contributor role on the specified resource group. Please work with the subscription owner to add Windows365 network interface contributor role assignment on the resource group for the Cloud PC service principal.|
```
Comment (Danielabom) [inline api-reference/beta/resources/cloudpconpremisesconnectionhealthcheck.md:94]: 1. Windows 365 will be mentioned correctly above. Please confirm if "Windows 365" is the accepted abbreviation.
2. Please remove the word "please."

**Review by Danielabom — CHANGES_REQUESTED** [review]
Thank you for working on this PR, @Sodaine. Please review my comments.

File: api-reference/beta/resources/cloudpconpremisesconnectionhealthcheck.md [hunk 5]
```diff
@@ -90,6 +90,8 @@ The result of a Cloud PC Azure network connection health check.
 |permissionCheckNoResourceGroupOwnerRole|The Cloud PC service principal doesn't have sufficient permissions on the Azure resource group. Please make sure that the Cloud PC service principal has the *Owner* permissions on the resource group. |
 |permissionCheckNoVNetContributorRole|The Cloud PC service principal doesn't have sufficient permissions on the Azure vNet. Please make sure that the Cloud PC service has the *Network contributor* permissions on the vNet.|
 |permissionCheckNoResourceGroupNetworkContributorRole|Cloud PC service principal doesn't have sufficient permissions on the Azure resource group. Please make sure that the  application has Network contributor permissions on the resource group.|
+|permissionCheckNoWindows365NetworkUserRole|Cloud PC service principal does not have Windows365 network user role on the specified virtual network. Please work with subscription owner to add the  Windows365 network user role assignment for the Cloud PC service principal.|
```
[the remark this answers is not in the document]
Comment (Emma0727) [reply api-reference/beta/resources/cloudpconpremisesconnectionhealthcheck.md]: Done, please help review again.

File: api-reference/beta/resources/cloudpconpremisesconnectionhealthcheck.md (hunk 4 above)
Comment (Emma0727) [reply api-reference/beta/resources/cloudpconpremisesconnectionhealthcheck.md]: Done, please help review again.

**Comment by Emma0727** [comment]
@Danielabom @Lauragra  Can you help review this PR when you are available?

**Pull request closed without merging**